joschahenningsen/TUM-Live
# No hardcoded file paths

### Motivation and Context


Closes #539
Throughout the worker there are multiple instances of hardcoded file paths. By using a PathProvider code duplications is mitigated and all the hardcoded endpoints can be easily found as they are all in once place.

### Description

All the hardcoded file paths I found in the worker module have been moved to the pathprovider. The path providing functions follow the OS agnostic style laid out by [filepath](https://pkg.go.dev/path/filepath)
The overall code has not been changed much, mostly just moved about, so the expected behaviour should also not have changed.

### Steps for Testing

Prerequisites:
- A machine running a Unix-based OS

1. Run go test github.com/TUM-Dev/gocast/tools/pathprovider
2. Ultimately these changes should incur no change in the behaviour of the overall project

### Screenshots

none
